src/query/src/dist_plan/analyzer.rs

```diff
@@ -155,7 +155,7 @@ impl DistPlannerAnalyzer {
             .collect::<DfResult<Vec<_>>>()?;
 
         // Some plans that are special treated (should not call `with_new_exprs` on them)
-        if !matches!(plan, LogicalPlan::Unnest(_)) {
+        if !matches!(plan, LogicalPlan::Unnest(_) | LogicalPlan::Join(_)) {
```

Comment:
is the join suppose to be here?

Reply:
@evenyag not sure how to fix it, can you give some hints?

datafusion requires the size of `exprs` must be at least two times of `on`'s size:

```rust
// inside datafusion's with_new_exprs method:

let equi_expr_count = on.len() * 2;
assert!(expr.len() >= equi_expr_count);
```

while the `exprs` seems very unlikely given it is produced by folding the `on`'s exprs then chaining some filters.

Reply:
it works, but are you sure to remove those codes?

Reply:
Yes, it was a workaround before https://github.com/apache/datafusion/pull/15055/